src/security/core/src/dds_security_serialize.c

```diff
@@ -742,7 +751,7 @@ DDS_Security_Deserialize_ParticipantBuiltinTopicData(
                 r = DDS_Security_Deserialize_DataHolder(dser, &pdata->permissions_token);
                 break;
             case PID_PROPERTY_LIST:
-                r = DDS_Security_Deserialize_PropertyQosPolicy(dser, &pdata->property);
+                r = DDS_Security_Deserialize_PropertyQosPolicy(dser, &pdata->property, len);
```

Comment:
Each policy should be deserialized based on the number of bytes remaining in the parameter, not just the properties. We have a (de)serializer pair for this stuff that works (ddsi_plist.c) and it really should build on that. I realise that takes a bit of effort because of those annoying DDS_Security_... types, but that effort is worth it just to get rid of having to maintain two implementations.

Having said that, that's something for a different PR.